src/wp-includes/class-wp-scripts.php

```diff
@@ -295,14 +295,48 @@ public function do_item( $handle, $group = false ) {
 		}
 
 		$before_handle = $this->print_inline_script( $handle, 'before', false );
-		$after_handle  = $this->print_inline_script( $handle, 'after', false );
 
 		if ( $before_handle ) {
 			$before_handle = sprintf( "<script%s id='%s-js-before'>\n%s\n</script>\n", $this->type_attr, esc_attr( $handle ), $before_handle );
 		}
 
-		if ( $after_handle ) {
-			$after_handle = sprintf( "<script%s id='%s-js-after'>\n%s\n</script>\n", $this->type_attr, esc_attr( $handle ), $after_handle );
+		$strategy = $this->get_eligible_loading_strategy( $handle );
+
+		$after_handle = '';
+
+		// Eligible loading strategies will only be 'async', 'defer', or ''.
+		if ( '' === $strategy ) {
+			$after_handle = $this->print_inline_script( $handle, 'after', false );
+
+			if ( $after_handle ) {
+				$after_handle = sprintf(
+					"<script%1\$s id='%2\$s-js-after'>\n%3\$s\n</script>\n",
+					$this->type_attr,
```

Comment:
could we add `esc_attr` here without breaking anything? Core is generally short on late escaping and I realize this wasn't escaped before but maybe worth considering.

Reply:
@adamsilverstein perhaps I am missing a secondary use case of `esc_attr()` here, but that is designed to escape attribute _values_ and not attribute keys, as it encodes `<, >, &, ” and ‘` [as per the docs](https://developer.wordpress.org/reference/functions/esc_attr/).

Give that we are outputting only an attribute key sans a value, I am not sure I see a reason for this.

Reply:
Also, since `$this->type_attr` is `private`, there doesn't seem to be any security risk (as the code stands now).